crates/red_knot_test/src/parser.rs

```diff
@@ -127,6 +129,118 @@ struct Section<'s> {
 #[newtype_index]
 struct EmbeddedFileId;
 
+#[derive(Debug)]
+pub(crate) struct CodeBlock<'s> {
+    pub(crate) code: &'s str,
+
+    /// The offset of the backticks beginning the code block within the markdown file
+    pub(crate) backtick_offset: TextSize,
+}
+
+/// Holds information about the position and the length of a single code block in a
+/// Markdown file.
+#[derive(Debug)]
+pub(crate) struct CodeBlockDimensions {
+    pub(crate) backtick_offset: TextSize,
+    pub(crate) line_count: usize,
+}
+
+/// Holds information about the position and length of all code blocks that are part of
+/// a single embedded file in a Markdown file. This is used to reconstruct absolute line
+/// numbers (in the Markdown file) from relative line numbers (in the embedded file).
+pub(crate) struct CodeBlockStructure {
+    start_line_and_line_count: Vec<(usize, usize)>,
+}
```

Comment:
I find the `Structure` postfix not very meaningful. Why isn't a `CodeBlock` a structure itself. 

Reading the comment, the focus here is that this structure holds the code blocks for a single file. 

How about: 

* EmbeddedFileLineIndex (has the advantage that everyone familiar with `LineIndex` immediately knows what this is)
* EmbeddedFileSourceMap (similar to the source map used for jupyter notebooks)
* FileCodeBlockLines